### PR TITLE
use max mon moves constant where able

### PR DESCRIPTION
**Pull request by melthelesbian**
im sure i didn't get all of them, but this should put a dent in

### CODE REVIEW

File: src/battle_ai_script_commands.c [hunk 1]
```diff
@@ -459,7 +459,7 @@ static u8 ChooseMoveOrAction_Doubles(void)
     s32 mostViableMovesNo;
     s16 mostMovePoints;
 
-    for (i = 0; i < 4; i++)
+    for (i = 0; i < MAX_MON_MOVES; i++)
```
[the remark this answers is not in the document]
Comment (melthelesbian) [reply src/battle_ai_script_commands.c]: thanks, will fix this one

File: src/battle_ai_script_commands.c [hunk 2]
```diff
@@ -1765,15 +1765,15 @@ static void BattleAICmd_if_has_move(void)
...
         {
             if (gBattleMons[sBattler_AI].moves[i] == *movePtr)
                 break;
         }
-        if (i == 4)
+        if (i == MAX_MON_MOVES)
             gAIScriptPtr += 8;
         else
-            gAIScriptPtr = T1_READ_PTR(gAIScriptPtr + 4);
+            gAIScriptPtr = T1_READ_PTR(gAIScriptPtr + MAX_MON_MOVES);
```
Comment (huderlem) [inline src/battle_ai_script_commands.c]: not correct

Comment (melthelesbian) [reply src/battle_ai_script_commands.c]: not even sure how that one got in there, i thought i hadn't done any of these

File: src/contest.c [hunk 3]
```diff
@@ -2635,7 +2635,7 @@ void sub_80DB584(void)
 {
     s32 i;
 
-    for (i = 0; i < 4; i++)
+    for (i = 0; i < MAX_MON_MOVES; i++)
```
Comment (huderlem) [inline src/contest.c]: not correct. This is the number of contest contestants

File: src/pokemon.c [hunk 4]
```diff
@@ -931,7 +931,7 @@ const u16 gSpeciesToNationalPokedexNum[] = // Assigns all species to the Nationa
     SPECIES_TO_NATIONAL(CHIMECHO),
 };
 
-const u16 gHoennToNationalOrder[] = // Assigns Hoenn Dex Pok�mon (Using National Dex Index)
+const u16 gHoennToNationalOrder[] = // Assigns Hoenn Dex Pok�mon (Using National Dex Index)
```
Comment (huderlem) [inline src/pokemon.c]: Two of the `é` characters are messed up.

Comment (melthelesbian) [reply src/pokemon.c]: idk how that happened, will try to figure out how to fix this

**Comment by melthelesbian** [comment]
Apologies for the errors here. Had to do these all by hand, and I guess there was bound to be a couple mistakes in the 3,000 or so lines of code I had to scan over for these

**Pull request merged**